pkg/lib/authenticationflow/declarative/utils_ldap.go

```diff
@@ -0,0 +1,50 @@
+package declarative
+
+import (
+	"github.com/authgear/authgear-server/pkg/api"
+	"github.com/authgear/authgear-server/pkg/api/model"
+	authflow "github.com/authgear/authgear-server/pkg/lib/authenticationflow"
+	"github.com/authgear/authgear-server/pkg/lib/authn/identity"
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/ldap"
+)
+
+func createIdentitySpecFromLDAPEntry(deps *authflow.Dependencies, serverName string, entry *ldap.Entry) (*identity.Spec, error) {
+	var ldapServerConfig *config.LDAPServerConfig
```

Comment:
Add a method in LDAPConfig to find a LDAPServerConfig. Just like OAuthSSOConfig.GetProviderConfig.

```
LDAPConfig.GetServerConfig(serverName string) (*LDAPServerConfig, bool)
```

And then LDAPFactory should take LDAPServerConfig instead of server_name. So it does not need to find twice.